src/Riok.Mapperly/Configuration/AttributeDataAccessor.cs

```diff
@@ -50,67 +50,103 @@ public IEnumerable<TAttribute> Access<TAttribute>(ISymbol symbol)
+        // with generic type parameters of the attribute class
+        // as ITypeSymbol parameters followed by all other parameters
+        // of the attribute constructor.
+        // Multiple attribute class constructors/generic data classes are not yet supported.
+        var argCount = typeArguments.Count + constructorArguments.Count;
+        foreach (var constructor in typeof(TData).GetConstructors())
+        {
+            var parameters = constructor.GetParameters();
+            if (parameters.Length != argCount)
+                continue;
+
+            var constructorArgumentValues = constructorArguments.Select(
+                (arg, i) => BuildArgumentValue(arg, parameters[i + typeArguments.Count].ParameterType)
+            );
+            var constructorTypeAndValueArguments = typeArguments.Concat(constructorArgumentValues).ToArray();
+            return (TData)Activator.CreateInstance(typeof(TData), constructorTypeAndValueArguments);
+        }
+
+        throw new InvalidOperationException($"{typeof(TData)} does not have a constructor with {argCount} parameters");
+    }
+
+    private static object? BuildArgumentValue(TypedConstant arg, Type targetType)
     {
         return arg.Kind switch
         {
             _ when arg.IsNull => null,
-            TypedConstantKind.Enum => GetEnumValue(arg),
-            TypedConstantKind.Array => BuildArrayValue(arg),
+            _ when targetType == typeof(StringMemberPath) => CreateMemberPath(arg),
+            TypedConstantKind.Enum => GetEnumValue(arg, targetType),
+            TypedConstantKind.Array => BuildArrayValue(arg, targetType),
             TypedConstantKind.Primitive => arg.Value,
-            TypedConstantKind.Type => arg.Value,
+            TypedConstantKind.Type when targetType == typeof(ITypeSymbol) => arg.Value,
             _
                 => throw new ArgumentOutOfRangeException(
-                    $"{nameof(AttributeDataAccessor)} does not support constructor arguments of kind {arg.Kind.ToString()}"
+                    $"{nameof(AttributeDataAccessor)} does not support constructor arguments of kind {arg.Kind.ToString()} or cannot convert it to {targetType}"
                 ),
         };
     }
 
-    private static object?[] BuildArrayValue(TypedConstant arg)
+    private static StringMemberPath CreateMemberPath(TypedConstant arg)
     {
-        var arrayTypeSymbol =
-            arg.Type as IArrayTypeSymbol
-            ?? throw new InvalidOperationException("Array typed constant is not of type " + nameof(IArrayTypeSymbol));
+        if (arg.Kind == TypedConstantKind.Array)
+        {
+            var values = arg.Values.Select(x => (string?)BuildArgumentValue(x, typeof(string))).WhereNotNull().ToList();
+            return new StringMemberPath(values);
+        }
+
+        if (arg is { Kind: TypedConstantKind.Primitive, Value: string v })
+        {
+            return new StringMemberPath(v.Split('.'));
```

Comment:
Extract the '.' into a utils or somewhere else to reuse it for the 'StringMemberPath' class.

Reply:
Fixed.